Makefile

```diff
@@ -125,11 +125,18 @@ OD			:=	${CROSS_COMPILE}objdump
 NM			:=	${CROSS_COMPILE}nm
 PP			:=	${CROSS_COMPILE}gcc -E
 
-ASFLAGS_aarch64		=	-mgeneral-regs-only
-TF_CFLAGS_aarch64	=	-mgeneral-regs-only -mstrict-align
+ifneq ($(findstring clang,$(notdir $(CC))),)
```

Comment:
The only reason we use findstring here is to support both CC=/foo/bar/clang and CC=/foo/bar/clang-X.Y

So base name of the path should normally be clang or clang-X.Y.